### DISCUSSION TITLE
How to size rotated control relative to parent?

### DISCUSSION THREAD

**Discussion by jtcyacomb** [AvaloniaUI/Avalonia#14756] [Q&A] [answered]
In this example, is it possible to size the blue panel relative to the red grid, as if the panel was not rotated?

I want the eventual vertical length of the panel to be the height of the grid minus the height of the text block.

The reason I want to do this is to rotate images, and have them fill the available space.

Example:

```axaml
<Window.Styles>
    <Style Selector="Panel.rotate">
        <Setter Property="RenderTransform">
            <Setter.Value>
                <RotateTransform Angle="90" />
            </Setter.Value>
        </Setter>
    </Style>
</Window.Styles>

<Grid Width="300" Height="150" Background="Red" RowDefinitions="*,Auto">
    <Panel Classes="rotate" Background="Blue" />
    <TextBlock Grid.Row="1" Text="Hello, world!" />
</Grid>
```

Result:

![image](https://github.com/AvaloniaUI/Avalonia/assets/153731734/2ede8fe0-cb9f-4239-a4d3-e7f0de5db7af)

**Comment by jp2masa** [comment, marked as answer]
You probably need to use [`LayoutTransformControl`](https://docs.avaloniaui.net/docs/reference/controls/layouttransformcontrol).

**Reply by jtcyacomb** [reply]
Thanks very much. It looks like it does exactly what I want.

```axaml
<Grid Width="300" Height="150" Background="Red" RowDefinitions="*,Auto">
    <LayoutTransformControl>
        <LayoutTransformControl.LayoutTransform>
            <RotateTransform Angle="90" />
        </LayoutTransformControl.LayoutTransform>

        <Panel Background="Blue">
            <TextBlock Text="Rotated text" />
        </Panel>
    </LayoutTransformControl>

    <TextBlock Grid.Row="1" Text="Hello, world!" />
</Grid>
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/153731734/cd984712-cd95-49e6-a018-0578b2e8665d)
